### ISSUE TITLE
React Series

### ISSUE THREAD

**Issue by duttakapil** [coderplex/meetups#1]
Let's discuss what our series of meetups on React ecosystem will include, what they should contain and how they should be formatted. These could be both online and/or offline meetups. 

### Ideas : 
- Working with React Higher Order Components
- How to test your react application
- How to route with React-Router
-  Lets talk about CSS-in-JS
- Intro to Redux
- Can we make our own Redux?
- Lets talk about Modern Javascript and dinosaurs
- Reading source code of React
- Building our own small library

**Comment by elevenpassin** [comment]
Also, how about these
- Different project structures and their advantages/disadvantages
- MobX
- Talks focussing in detail on certain parts of React, like Controlled Components, Portals, Escape Hatches, Performance, and best practices.
- And when coming to redux, talks in detail on topics like Immutability, Container, and Presentational Components.
- Redux source code is also very short and simple. Maybe a talk going through it will be great. :D 


Maybe React Native needs its own section.

**Comment by duttakapil** [comment]
@buoyantair good suggestions. Let's wait for other React devs from our community, like @M-ZubairAhmed to respond with their thoughts

**Comment by duttakapil** [comment]
Topic : React 101
Date : 16th December
Timings : 10 am - 1 pm
Speakers : Zubair and Aravind
Venue : T-Hub 
Status : [Announced](https://www.meetup.com/coderplex/events/245825655/)

**Comment by duttakapil** [comment]
Topic : Walking through Coderplex code base
Date : ?
Timings : ?
Speakers : @M-ZubairAhmed 
Venue : Online Web Cast
Status : Still planning

CC : @M-ZubairAhmed

**Comment by duttakapil** [comment]
Topic : Reading React Source Code
Date : ?
Timings : ?
Speakers : @M-ZubairAhmed, ?
Venue : Not sure (online or offline)
Status : Still planning

**Comment by M-ZubairAhmed** [comment]
Topic : Walking through Coderplex code base
Date :  13 Dec '17
Timings : 6:00 - 7:00 pm
Speakers : @M-ZubairAhmed, @pbteja1998
Venue : Online Web Cast
Status : Confirmed

**Comment by duttakapil** [comment]
@M-ZubairAhmed will one hour be sufficient? Details of the format of the webcast?

**Comment by M-ZubairAhmed** [comment]
yes for start, if we have lots of question then we can think about it. Starting with the frontend code base then backend

**Comment by duttakapil** [comment]
Ok. Starting at 6 pm or 7 pm?

**Comment by M-ZubairAhmed** [comment]
6 i updated comment

**Comment by elevenpassin** [comment]
[UPDATE]
- Cover Fragments, Error boundaries
- Flow or Typescript for static typing! 
- How about things like Styled components
- SSR (Something I haven't learnt still...... D:)
- Ant.desgin, semantic-ui react, react-bootstrap, bootstrap, material-ui, material, foundation, BEM?, what about Bulma JS? CSS? SASS? OMG there's so many options to style help me choose!
- Again, just like js, react ecosystem is freakin FAT, even for animations you have react-motion, css animations, js based animations, velocity react, the react transition group... Lots of lots of options for components, loads of component libraries, and options for styling and animation. There should be a talk on how to cope up with this react env too 😂

**Comment by duttakapil** [comment]
We cannot practically cover any and everything in the React ecosystem, the idea is to only cover things that are harder to understand and learn on your own. Beyond that, if any community member is interested, they can come forward to host and lead meetups with us. These could be online or even offline.

**Comment by elevenpassin** [comment]
@duttakapil Noooo I meant that the JS and React eco system is very huge, and sometimes it's terrible. I think there should be a series of talks on a single meetup by people in the community, on how they are keeping on the edge, learning new frameworks etc.

**Comment by duttakapil** [comment]
@buoyantair the problem with that is, in the developer ecosystem of Hyderabad, we do not have sufficient people who use these technologies to build things, so we wouldn't have enough people to come down and discuss how they are being on the edge, staying up to date. In other words, the developer culture in the city is poor and we need to work on that
